Fix bug in FlxSound.proximity

Dovyski opened the pull request:
FlxSound.proximity was inverted: the closer the object was, the lower was the volume.
Fix https://github.com/FlixelCommunity/flixel/issues/4, https://github.com/AdamAtomic/flixel/issues/228 .

Dovyski committed:
Fix bug in FlxSound.proximity

FlxSound.proximity was inverted: the closer the object was, the lower was the volume.
Fix https://github.com/FlixelCommunity/flixel/issues/4, https://github.com/AdamAtomic/flixel/issues/228 .

moly commented:
I think `var d:Number = (_target.x-x)/_radius;` should be swapped to `var d:Number = (x-_target.x)/_radius;`. If the target object is to the left of you the sound should be coming from the left speaker.

Dovyski commented:
Ok, I will fix that.

Dovyski commented:
I've tested the code and the original version `_target.x - x` works the way you described. If the target object is to the left of you, then the result is negative, what produces left-panning.

According to the `SoundTransform` [docs](http://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/media/SoundTransform.html#pan), the `pan` property is:

> The left-to-right panning of the sound, ranging from -1 (full pan left) to 1 (full pan right).
> A value of 0 represents no panning (balanced center between right and left).

moly commented:
Sorry, I meant to say that if the **FlxSound** is to the left of the target object, the sound should becoming from the left speaker.

Dovyski commented:
I was just wondering if the sound source is to the left of the target, the target will hear the sound source using its **right** ear, isn't it? This is the current behavior of `FlxSound`.

Do you disagree about that, @moly ?

moly commented:
Here's how I picture it:
![Mario](http://i.imgur.com/n10g5.jpg)

Imagine that Mario is the target, and that the Goomba is producing a sound. In this 2D world the Goomba is behind Mario and shown on the left side of the screen, so you would expect the sound to come from behind him on the left side speaker.

Dovyski commented:
Ok, now I got it! :) I couldn't agree more. I will change that line according to your suggestion.

Dovyski committed:
Pan adjustments

If FlxSound is to the left of the target object, the sound comes from the left speaker.

IQAndreas commented:
``` as3
var d:Number = (x-_target.x)/_radius;
```

Sorry, correct me if I'm wrong, but the way you had it was right, correct?

If we use the above example, the location of Mario would be `x=100` and the location of the goomba `_target.x=20`. This means `100-20=80`, positive, so the sound would be coming out of the right speaker.

moly commented:
In the above example Mario is the `_target`, and the Goomba would be `x` and `y`. The Goomba is emitting the sound.

IQAndreas commented:
> In the above example Mario is the _target, and the Goomba would be x and y. The Goomba is emitting the sound.
> 
> _(from `FlxSound)_ The X position of this **sound** in world coordinates.

Ah, yes, I see now. Perfect, then I'll go ahead and merge this one.
